scrapers/wv/events.py

```diff
@@ -132,17 +128,40 @@ def scrape_meeting_page(self, url):
                 agenda = event.add_agenda_item(
                     row.text_content().strip().replace("\u25a1", "")
                 )
-                for bill in re.findall(self.bill_regex, row.text_content()):
+
+                # Matches (SJR, HCR, HB, HR, SCR, SB, HJR, SR) + id
+                # Allows for house, senate, joint, or bill to be fully spelled out
+                # Allows for "." after H, S, J, C, and B
+                # Allows for up to two spaces before the id
+                bills = re.findall(
+                    r"((S\.?|Senate|H\.?|House)\s?((J|C|Joint)\.?\s?)?(B\.?|Bill|R\.?)\s?\s?(\d+))",
+                    row.text_content(),
+                    flags=re.IGNORECASE,
+                )
+
+                for bill in bills:
                     bill_id = re.sub(r"\.\s*", "", bill[0], flags=re.IGNORECASE)
                     bill_id = re.sub(r"house bill", "HB", bill_id, flags=re.IGNORECASE)
                     bill_id = re.sub(r"senate bill", "SB", bill_id, flags=re.IGNORECASE)
+
+                    # Final step to set correct number of spaces in the id
+                    components = re.search(
```

Comment:
It's possibly a premature optimization, but typically we want to pre-compile regexes if we're going to use them in loops:

```
component_re = re.compile(r"([A-Z]+)\s*(\d+)")
for bill in bills:
    components = component_re.search(bill_id, re.IGNORECASE)
```

And so on.

Reply:
If it would be a pain to fix all these regexes, don't let it stop a merge. But something to consider generally.

Reply:
Fixed with https://github.com/openstates/openstates-scrapers/pull/4521/commits/d4f0ae7ef37b73579fccc2cb962f36619d7688da

Reply:
@a-cat-named-snowball for the sake of consistency, can you do the same for those three patterns on lines 145-147, that is, pre-compile them before the for loop?